### PR TITLE
Tutorial

### PR DESCRIPTION
**Pull request by B4CKSL4SH-44**

### CODE REVIEW

**Comment by B4CKSL4SH-44** [comment]
This PR ist just a savepoint, I have to split my tutorial in different levels due to the complexity...

File: components/BuildingMenu.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  width: 95%;
+  color: white;
+  background-color: rgba(0, 0, 0, 0.5);
+  transition: 0.5s;
+  ${props => (props.buildingMenuState ? "" : "opacity: 0; z-index: -1;")}
+`;
+
+const BuildingMenuHeadline = styled.h1`
+  text-align: center;
+  margin-top: 10px;
```
Comment (stefankorder) [inline components/BuildingMenu.js]: Better use relative units at this point.

File: components/BuildingMenu.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`;
+
+const InputContainer = styled.div`
+  border-radius: 20px;
+  box-shadow: 0 0 8px white;
+  background-color: rgba(0, 0, 0, 0.6);
+  padding: 5px;
```
Comment (stefankorder) [inline components/BuildingMenu.js]: Better use relative units at this point.

File: components/BuildingMenu.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin: 5px auto;
```
Comment (stefankorder) [inline components/BuildingMenu.js]: Better use relative units at this point.

File: components/BuildingMenu.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+`;
+
+const InputElement = styled.input`
+  margin: 15px;
```
Comment (stefankorder) [inline components/BuildingMenu.js]: Better use relative units at this point.

File: components/BuildingMenu.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  transform: scale(1.5);
+`;
+
+const Button = styled.button`
+  margin: 10px;
```
Comment (stefankorder) [inline components/BuildingMenu.js]: Better use relative units at this point.

File: components/OverlaySmall.js [hunk 6]
```diff
@@ -19,13 +19,14 @@ export default function OverlaySmall({
   );
 }
 const Overlay = styled.div`
-  width: 95%;
+  width: 98%;
+  position: absolute;
   font-size: 20px;
   padding: 30px;
```
Comment (stefankorder) [inline components/OverlaySmall.js]: Better use relative units at this point.

File: components/OverlaySmall.js [hunk 7]
```diff
@@ -19,13 +19,14 @@ export default function OverlaySmall({
   );
 }
 const Overlay = styled.div`
-  width: 95%;
+  width: 98%;
+  position: absolute;
   font-size: 20px;
```
Comment (stefankorder) [inline components/OverlaySmall.js]: Better use relative units at this point.

File: components/Tasks.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    </TasksContainer>
+  );
+}
+
+const TasksContainer = styled.div`
+  width: 95%;
+  font-size: 18px;
+  background-color: rgba(0, 255, 255, 0.3);
+  border-radius: 20px;
+  padding: 10px;
```
Comment (stefankorder) [inline components/Tasks.js]: Better use relative units at this point.

File: components/Tasks.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+      Tasks:
+      <br />
+      {children}
+    </TasksContainer>
+  );
+}
+
+const TasksContainer = styled.div`
+  width: 95%;
+  font-size: 18px;
```
Comment (stefankorder) [inline components/Tasks.js]: Better use relative units at this point.

File: functions/cutTrees.js [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+  //         item.color === "forest" &&
+  //         array[item.id + 10].color === "lumberhut"
+  //       ) {
+  //         choppedWood++;
+  //         return {...item, color: "grass"};
+  //       } else {
+  //         return item;
+  //       }
+  //     })
+  //   );
```
Comment (stefankorder) [inline functions/cutTrees.js]: Please delete this unused lines of code.

File: pages/level1/index.js [hunk 11]
```diff
@@ -1,34 +1,158 @@
...
   const [overlayState, setOverlayState] = useState(true);
-  const text1 =
-    "Welcome to the island. Your first task is to build a lumberhut. Simply click on the sand tile, choose the lumberhut from the menu and build!";
+  const [textState, setTextState] = useState(1);
+  const [timeoutState, setTimeoutState] = useState(0);
+  const [pumpWaterState, setPumpWaterState] = useState(false);
+  const [possibleBuildings, setPossibleBuildings] = useState([
+    {name: "lumberhut", price: "wood x2"},
+    // {name: "house", price: ""},
+    //{name: "tower", price: "wood x4"}
```
Comment (stefankorder) [inline pages/level1/index.js:25]: Please delete this unused lines of code.

**Review by stefankorder — CHANGES_REQUESTED** [review]
Good work! Please take a look at my comments.

File: pages/level1/index.js (hunk 11 above)
Comment (B4CKSL4SH-44) [reply pages/level1/index.js:25]: Those are notes for later use, i kinda use this code as a template for other levels....

**Review by stefankorder — APPROVED** [review]
LGTM! Please check your code if you could use more relative units in some cases.

**Pull request merged**